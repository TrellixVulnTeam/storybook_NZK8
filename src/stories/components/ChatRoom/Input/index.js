// @flow

import React, { PureComponent } from 'react';
import { Picker } from 'emoji-mart';
import fileUpload from 'utils/fileUpload';
import uniqueId from 'lodash.uniqueid';
import * as Actions from 'store/Actions';
import {
  Wrapper,
  TextArea,
  FileButton,
  PaperclipIcon,
  ThumbnailContainer,
  ThumbnailPreview,
  EmojiWrapper,
  EmojiIconActive,
  EmojiIconInactive,
  UploadingImage,
  Error,
} from './styled';

const validExtensions = ['jpg', 'jpeg', 'png', 'gif'];

type Props = {
  t: Function,
  message: string,
  images: Array<Object>,
  imagesUploading: boolean,
  singleUpload?: boolean,
  pickerStyle?: Object,
  disabled?: boolean,
  disableEnterSubmit?: boolean,
  onImagesMutate: (Array<Object>) => *,
  onChange: string => *,
  onSubmit: () => *,
};

type State = {
  emojiKeyboard: boolean,
  error: ?string,
};

class ChatRoomInput extends PureComponent<Props, State> {
  inputForm = null;

  textarea = null;

  // eslint-disable-next-line react/state-in-constructor
  state = {
    emojiKeyboard: false,
    error: null,
  };

  /**
   * Handle sizing of textArea
   * @param {Object} prevProps
   */
  componentDidUpdate() {
    this.handleGrow();
  }

  processSelectedFiles = (files: Array<File>) => {
    const validFiles = [];
    let invalidFiles = 0;
    // eslint-disable-next-line no-restricted-syntax
    for (const file of files) {
      const extension = file.name
        .split('.')
        .pop()
        .toLowerCase();
      if (validExtensions.indexOf(extension) === -1) {
        invalidFiles += 1;
        // eslint-disable-next-line no-continue
        continue;
      }
      this.readFile(file);
      validFiles.push({
        _id: uniqueId('thumb-'),
        file,
      });
    }

    if (invalidFiles) {
      this.setState({
        error:
          `${invalidFiles} unsupported file${invalidFiles === 1 ? '' : 's'} ignored! ` +
          `Supported types are: ${validExtensions.join(', ')}`,
      });
    }

    if (validFiles.length) {
      const images = this.props.images.concat(validFiles);
      this.props.onImagesMutate(images);
    }
  };

  readFile = async (file: File) => {
    try {
      const dataUrl = await fileUpload.readAsDataUrl(file);
      const images = this.props.images.slice(0);
      images.forEach(i => {
        if (i.file === file) {
          // eslint-disable-next-line no-param-reassign
          i.dataUrl = dataUrl;
        }
      });
      this.props.onImagesMutate(images);
    } catch (e) {
      this.setState({
        error: `Error reading file!`,
      });
    }
  };

  handleFileSelected = (e: Object) => {
    const { files } = e.currentTarget;
    if (this.props.singleUpload && files.length > 1) {
      Actions.alertModal({
        id: 1,
        maxWidth: 522,
        header: 'Max upload of one file per comment',
        body: 'Please select one file to upload.',
        onConfirm: () => Actions.removeModal(1),
        confirmLabel: 'Close',
        cancelLabel: 'cancel',
      });
      e.preventDefault();
      return;
    }
    this.processSelectedFiles(files);
  };

  handleDeleteImage = (imageId: string) => {
    const images = this.props.images.slice(0);
    images.forEach(i => {
      // eslint-disable-next-line no-underscore-dangle
      if (i._id === imageId) {
        // eslint-disable-next-line no-param-reassign
        i.deleted = true;
        // eslint-disable-next-line no-param-reassign
        i.file = null;
        // eslint-disable-next-line no-param-reassign
        i.dataUrl = null;
      }
    });

    this.props.onImagesMutate(images);
  };

  handleToggleEmojiKeyboard = () => {
    this.setState({
      // eslint-disable-next-line react/no-access-state-in-setstate
      emojiKeyboard: !this.state.emojiKeyboard,
    });
  };

  handlePickEmoji = (emoji: Object) => {
    this.props.onChange(`${this.props.message}${emoji.native}`);
  };

  handleGrow = () => {
    if (!this.textarea) {
      return;
    }

    if (!this.props.message) {
      // Needed to reset textarea height
      this.textarea.style.height = null;
    } else {
      this.textarea.style.height = `${this.textarea.scrollHeight}px`;
    }
  };

  handleEnterPress = (e: Object) => {
    if (e.keyCode === 13 && e.shiftKey === false && this.props.disableEnterSubmit !== true) {
      e.preventDefault();
      this.setState({ error: null });
      this.props.onSubmit();
    }
  };

  /**
   * Renders the thumbnail previews for selected images
   * @return {React$Node}
   */
  renderImages() {
    const thumbnails = this.props.images.map(item => {
      if (item.deleted) {
        return null;
      }

      return (
        // eslint-disable-next-line no-underscore-dangle
        <ThumbnailPreview key={item._id}>
          <span className="thumbnail" style={{ backgroundImage: `url(${item.dataUrl})` }} />
          {/* eslint-disable-next-line jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */}
          <span
            className="delete-image"
            onClick={
              // eslint-disable-next-line no-underscore-dangle
              () => this.handleDeleteImage(item._id)
            }
          >
            Ã—
          </span>
          {this.props.imagesUploading ? (
            <UploadingImage>
              <div />
              <div />
              <div />
            </UploadingImage>
          ) : null}
        </ThumbnailPreview>
      );
    });

    return <ThumbnailContainer>{thumbnails}</ThumbnailContainer>;
  }

  /**
   * Renders the input text box for the chat room view
   * @return {React$Node}
   */
  render() {
    const imageCount = this.props.images.reduce((sum, image) => {
      return sum + !image.deleted;
    }, 0);

    return (
      // eslint-disable-next-line no-return-assign
      <form ref={comp => (this.inputForm = comp)}>
        <Wrapper>
          <FileButton>
            <PaperclipIcon />
            <input
              type="file"
              onChange={this.handleFileSelected}
              value=""
              accept="image/*"
              multiple
            />
          </FileButton>
          <TextArea
            type="text"
            // eslint-disable-next-line no-return-assign
            innerRef={r => (this.textarea = r)}
            autoFocus
            value={this.props.message}
            onChange={e => this.props.onChange(e.target.value)}
            placeholder={this.props.t('pages.messages.writeMessagePlaceholder')}
            onKeyDown={this.handleEnterPress}
            disabled={this.props.disabled}
          />
          {this.state.emojiKeyboard ? (
            <EmojiIconActive onClick={this.handleToggleEmojiKeyboard} />
          ) : (
            <EmojiIconInactive onClick={this.handleToggleEmojiKeyboard} />
          )}
          {this.state.emojiKeyboard && (
            <EmojiWrapper>
              <Picker
                onSelect={this.handlePickEmoji}
                style={this.props.pickerStyle && this.props.pickerStyle}
              />
            </EmojiWrapper>
          )}
        </Wrapper>
        {imageCount ? <div>{this.renderImages()}</div> : null}
        {this.state.error ? <Error>{this.state.error}</Error> : null}
      </form>
    );
  }
}

export default ChatRoomInput;
