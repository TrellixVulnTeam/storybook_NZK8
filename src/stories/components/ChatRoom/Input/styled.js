// @flow

import styled, { keyframes } from 'styled-components';
import { colors } from 'utils/StyleGuide';
import paperclip from './img/paperclip.svg';
import emojiIconActive from './img/emojiIconActive.svg';
import emojiIconInactive from './img/emojiIconInactive.svg';

export const Wrapper = styled.div`
  position: relative;
  display: flex;
  flex-flow: row nowrap;
  align-items: center;
  border-top: 1px solid ${colors.paleGrey};
`;

export const TextArea = styled.textarea`
  flex: 1;
  font-size: 14px;
  color: ${colors.charcoalGrey};
  padding: 20px 45px;
  border: none;
  outline: none;
  resize: none;
  overflow: hidden;

  ::placeholder {
    color: ${colors.steel};
  }
`;

export const FileButton = styled.div`
  position: absolute;
  left: 10px;
  overflow: hidden;
  cursor: pointer;

  input[type='file'] {
    position: absolute;
    top: 0;
    right: 0;
    min-width: 100%;
    min-height: 100%;
    font-size: 100px;
    text-align: right;
    filter: alpha(opacity=0);
    opacity: 0;
    outline: none;
    background: white;
    cursor: inherit;
    display: block;
  }
`;

export const PaperclipIcon = styled(paperclip)`
  width: 25px;
  height: 25px;
`;

export const ThumbnailContainer = styled.div`
  display: flex;
  padding: 10px;
  width: 100%;
  overflow: hidden;
`;

export const ThumbnailPreview = styled.div`
  width: 20%;
  position: relative;

  .thumbnail {
    display: block;
    padding-bottom: 85%;
    height: 0;
    margin: 2px;
    background-position: center;
    background-size: cover;
    border-radius: 6px;
  }

  .delete-image {
    position: absolute;
    right: 5px;
    top: 10px;
    font-size: 25px;
    cursor: pointer;
    color: ${colors.strawberry};
    line-height: 0px;
    font-weight: bold;
  }
`;

export const EmojiWrapper = styled.div`
  position: absolute;
  right: 0;
  bottom: 50px;
`;

export const EmojiIconActive = styled(emojiIconActive)`
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  cursor: pointer;
`;

export const EmojiIconInactive = styled(emojiIconInactive)`
  position: absolute;
  width: 25px;
  height: 25px;
  right: 10px;
  cursor: pointer;
`;

export const Error = styled.span`
  color: ${colors.strawberry};
  font-size: 12px;
`;

const bounceAnimation = keyframes`
  0%, 80%, 100% {
    transform: scale(0); }
  40% {
    transform: scale(1);
  }
`;

export const UploadingImage = styled.div`
  position: absolute;
  top: -2px;
  left: -2px;
  text-align: center;
  padding-top: 35%;
  background-color: rgba(0, 0, 0, 0.8);
  margin: 2px;
  height: 100%;
  width: 100%;
  border-radius: 6px;

  div {
    width: 25px;
    height: 25px;
    background-color: ${colors.steel};
    border-radius: 100%;
    display: inline-block;
    animation: ${bounceAnimation} 1.4s ease-in-out 0s infinite both;
  }

  div:first-child {
    animation-delay: -0.32s;
  }

  div:nth-child(2) {
    animation-delay: -0.16s;
  }
`;
