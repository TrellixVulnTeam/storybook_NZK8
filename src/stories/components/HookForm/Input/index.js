// @flow
import React from 'react';
import Label from '../Label';
import FormGroup from '../FormGroup';
import ErrorMessage from '../ErrorMessage';
import InputElem from './styled';

type Props = {
  label?: string,
  name: string,
  inputRef: Function,
  type: string,
  required: boolean,
  error?: string | object,
  autoComplete?: string,
  autoFocus?: boolean,
  min?: number,
};

const HookInput = (props: Props) => {
  return (
    <FormGroup error={!!(props.error && props.error.message)}>
      {props.label && <Label label={props.label} required={props.required} />}
      <InputElem
        name={props.name}
        innerRef={props.inputRef}
        type={props.type}
        autoComplete={props.autoComplete || 'off'}
        autoFocus={props.autoFocus}
        min={props.min}
      />
      {props.error && <ErrorMessage error={props.error.message} />}
    </FormGroup>
  );
};

export default HookInput;
