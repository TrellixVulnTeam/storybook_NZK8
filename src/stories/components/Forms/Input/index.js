// @flow

import React from 'react';
import { Field } from 'formik';
import { Wrapper, CustomError } from './styled';

// eslint-disable-next-line no-unused-vars
type Props = {
  type?: 'text' | 'email' | 'number',
  error?: string,
  [key: string]: any,
};

/**
 * Render a react-select input, for use with
 * Formik's <Field> component or as a standalone component.
 * @param {Object} props
 * @return {React$Node}
 */
export const InputComponent = ({ error, ...props }: $FlowLintFix) => (
  <input
    {...props}
    onBlur={(...args) => {
      props.form.setFieldTouched(props.field.name, true);

      if (typeof props.onBlur === 'function') {
        props.onBlur(...args);
      }
    }}
    onKeyDown={(e, ...args) => {
      if (e.key === 'Enter' && typeof props.form.handleSubmit === 'function') {
        props.form.handleSubmit(e, ...args);
      }
    }}
    onChange={(e, ...args) => {
      props.form.setFieldValue(props.field.name, e.target.value);

      if (typeof props.onChange === 'function') {
        props.onChange(e.target.value, ...args);
      }
    }}
    value={props.value !== undefined ? props.value : props.field.value}
  />
);

/**
 * @param {Object} fieldProps
 * @return {React$Node}
 */
const WrappedInput = (fieldProps: Object) => {
  const { isValidatedBySchema } = fieldProps;
  const { name } = fieldProps.field;
  const error = fieldProps.form.errors[name];

  return (
    <Wrapper error={error}>
      <InputComponent {...fieldProps} />
      {isValidatedBySchema && !!error && <CustomError>{error}</CustomError>}
    </Wrapper>
  );
};

export default (props: Props) => <Field {...props} component={WrappedInput} />;
