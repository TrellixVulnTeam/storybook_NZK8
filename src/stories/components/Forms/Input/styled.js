// @flow
/* eslint-disable import/prefer-default-export */

import styled from 'styled-components';
import { colors } from 'utils/StyleGuide';
import { inputStyle } from '../FormStyles';

export const Wrapper = styled.div`
  position: relative;
  width: 100%;

  input {
    ${inputStyle};
    padding: 0 10px;
  }
`;

export const CustomError = styled.div`
  width: 330px;
  height: 20px;
  font-size: 14px;
  color: ${colors.strawberry};
  display: contents;
`;
