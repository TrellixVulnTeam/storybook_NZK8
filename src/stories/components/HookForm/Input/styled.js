// @flow
import styled from 'styled-components';

const InputElem = styled.input`
  display: flex;
  flex: 1;
  border: 1px solid ${p => p.theme.colors.silver};
  outline: none;
  width: 100%;
  height: 40px;
  font-size: 14px;
  color: ${p => p.theme.colors.darkGrey};
  border-radius: 4px;
  padding: 10px;
  &:focus{
    border-color: ${p => p.theme.colors.brightBlue};
  }
`;

export default InputElem;