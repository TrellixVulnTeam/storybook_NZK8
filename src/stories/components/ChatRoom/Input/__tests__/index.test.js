import React from 'react';
import { mount } from 'enzyme';
import Input from 'components/ChatRoom/Input/index';

describe('<Input />', () => {
  const mandatoryProps = {
    t: jest.fn(),
    message: 'Message',
    images: [{}],
    imagesUploading: true,
    onImagesMutate: jest.fn(),
    onChange: jest.fn(),
    onSubmit: jest.fn(),
  };

  it('should render correctly with mandatory props', () => {
    const sut = mount(<Input {...mandatoryProps} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly with deleted image', () => {
    const props = {
      ...mandatoryProps,
      images: [
        {
          _id: 0,
          file: '',
          dataUrl: '',
          deleted: true,
        },
        {
          _id: 1,
          file: '',
          dataUrl: '',
          deleted: false,
        },
      ],
    };
    const sut = mount(<Input {...props} />);

    expect(sut).toMatchSnapshot();
  });

  it('should render correctly without images uploading', () => {
    const props = {
      ...mandatoryProps,
      imagesUploading: false,
    };
    const sut = mount(<Input {...props} />);

    expect(sut).toMatchSnapshot();
  });
});
