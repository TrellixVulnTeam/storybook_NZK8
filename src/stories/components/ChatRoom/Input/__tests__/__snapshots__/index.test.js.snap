// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> should render correctly with deleted image 1`] = `
<ChatRoomInput
  images={
    Array [
      Object {
        "_id": 0,
        "dataUrl": "",
        "deleted": true,
        "file": "",
      },
      Object {
        "_id": 1,
        "dataUrl": "",
        "deleted": false,
        "file": "",
      },
    ]
  }
  imagesUploading={true}
  message="Message"
  onChange={[MockFunction]}
  onImagesMutate={[MockFunction]}
  onSubmit={[MockFunction]}
  t={
    [MockFunction] {
      "calls": Array [
        Array [
          "pages.messages.writeMessagePlaceholder",
        ],
        Array [
          "pages.messages.writeMessagePlaceholder",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <form>
    <styled__Wrapper>
      <div
        className="styled__Wrapper-pnin0k-0 gnifIB"
      >
        <styled__FileButton>
          <div
            className="styled__FileButton-pnin0k-2 ksWiUG"
          >
            <styled__PaperclipIcon>
              <test-file-stub
                className="styled__PaperclipIcon-pnin0k-3 cVimtJ"
              />
            </styled__PaperclipIcon>
            <input
              accept="image/*"
              multiple={true}
              onChange={[Function]}
              type="file"
              value=""
            />
          </div>
        </styled__FileButton>
        <styled__TextArea
          autoFocus={true}
          innerRef={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="Message"
        >
          <textarea
            autoFocus={true}
            className="styled__TextArea-pnin0k-1 eLkOrk"
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="Message"
          />
        </styled__TextArea>
        <styled__EmojiIconInactive
          onClick={[Function]}
        >
          <test-file-stub
            className="styled__EmojiIconInactive-pnin0k-8 fXbUEp"
            onClick={[Function]}
          />
        </styled__EmojiIconInactive>
      </div>
    </styled__Wrapper>
    <div>
      <styled__ThumbnailContainer>
        <div
          className="styled__ThumbnailContainer-pnin0k-4 dtGwbz"
        >
          <styled__ThumbnailPreview
            key="1"
          >
            <div
              className="styled__ThumbnailPreview-pnin0k-5 heydDS"
            >
              <span
                className="thumbnail"
                style={
                  Object {
                    "backgroundImage": "url()",
                  }
                }
              />
              <span
                className="delete-image"
                onClick={[Function]}
              >
                ×
              </span>
              <styled__UploadingImage>
                <div
                  className="styled__UploadingImage-pnin0k-10 jJvdND"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </styled__UploadingImage>
            </div>
          </styled__ThumbnailPreview>
        </div>
      </styled__ThumbnailContainer>
    </div>
  </form>
</ChatRoomInput>
`;

exports[`<Input /> should render correctly with mandatory props 1`] = `
<ChatRoomInput
  images={
    Array [
      Object {},
    ]
  }
  imagesUploading={true}
  message="Message"
  onChange={[MockFunction]}
  onImagesMutate={[MockFunction]}
  onSubmit={[MockFunction]}
  t={
    [MockFunction] {
      "calls": Array [
        Array [
          "pages.messages.writeMessagePlaceholder",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <form>
    <styled__Wrapper>
      <div
        className="styled__Wrapper-pnin0k-0 gnifIB"
      >
        <styled__FileButton>
          <div
            className="styled__FileButton-pnin0k-2 ksWiUG"
          >
            <styled__PaperclipIcon>
              <test-file-stub
                className="styled__PaperclipIcon-pnin0k-3 cVimtJ"
              />
            </styled__PaperclipIcon>
            <input
              accept="image/*"
              multiple={true}
              onChange={[Function]}
              type="file"
              value=""
            />
          </div>
        </styled__FileButton>
        <styled__TextArea
          autoFocus={true}
          innerRef={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="Message"
        >
          <textarea
            autoFocus={true}
            className="styled__TextArea-pnin0k-1 eLkOrk"
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="Message"
          />
        </styled__TextArea>
        <styled__EmojiIconInactive
          onClick={[Function]}
        >
          <test-file-stub
            className="styled__EmojiIconInactive-pnin0k-8 fXbUEp"
            onClick={[Function]}
          />
        </styled__EmojiIconInactive>
      </div>
    </styled__Wrapper>
    <div>
      <styled__ThumbnailContainer>
        <div
          className="styled__ThumbnailContainer-pnin0k-4 dtGwbz"
        >
          <styled__ThumbnailPreview>
            <div
              className="styled__ThumbnailPreview-pnin0k-5 heydDS"
            >
              <span
                className="thumbnail"
                style={
                  Object {
                    "backgroundImage": "url(undefined)",
                  }
                }
              />
              <span
                className="delete-image"
                onClick={[Function]}
              >
                ×
              </span>
              <styled__UploadingImage>
                <div
                  className="styled__UploadingImage-pnin0k-10 jJvdND"
                >
                  <div />
                  <div />
                  <div />
                </div>
              </styled__UploadingImage>
            </div>
          </styled__ThumbnailPreview>
        </div>
      </styled__ThumbnailContainer>
    </div>
  </form>
</ChatRoomInput>
`;

exports[`<Input /> should render correctly without images uploading 1`] = `
<ChatRoomInput
  images={
    Array [
      Object {},
    ]
  }
  imagesUploading={false}
  message="Message"
  onChange={[MockFunction]}
  onImagesMutate={[MockFunction]}
  onSubmit={[MockFunction]}
  t={
    [MockFunction] {
      "calls": Array [
        Array [
          "pages.messages.writeMessagePlaceholder",
        ],
        Array [
          "pages.messages.writeMessagePlaceholder",
        ],
        Array [
          "pages.messages.writeMessagePlaceholder",
        ],
      ],
      "results": Array [
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
        Object {
          "type": "return",
          "value": undefined,
        },
      ],
    }
  }
>
  <form>
    <styled__Wrapper>
      <div
        className="styled__Wrapper-pnin0k-0 gnifIB"
      >
        <styled__FileButton>
          <div
            className="styled__FileButton-pnin0k-2 ksWiUG"
          >
            <styled__PaperclipIcon>
              <test-file-stub
                className="styled__PaperclipIcon-pnin0k-3 cVimtJ"
              />
            </styled__PaperclipIcon>
            <input
              accept="image/*"
              multiple={true}
              onChange={[Function]}
              type="file"
              value=""
            />
          </div>
        </styled__FileButton>
        <styled__TextArea
          autoFocus={true}
          innerRef={[Function]}
          onChange={[Function]}
          onKeyDown={[Function]}
          type="text"
          value="Message"
        >
          <textarea
            autoFocus={true}
            className="styled__TextArea-pnin0k-1 eLkOrk"
            onChange={[Function]}
            onKeyDown={[Function]}
            type="text"
            value="Message"
          />
        </styled__TextArea>
        <styled__EmojiIconInactive
          onClick={[Function]}
        >
          <test-file-stub
            className="styled__EmojiIconInactive-pnin0k-8 fXbUEp"
            onClick={[Function]}
          />
        </styled__EmojiIconInactive>
      </div>
    </styled__Wrapper>
    <div>
      <styled__ThumbnailContainer>
        <div
          className="styled__ThumbnailContainer-pnin0k-4 dtGwbz"
        >
          <styled__ThumbnailPreview>
            <div
              className="styled__ThumbnailPreview-pnin0k-5 heydDS"
            >
              <span
                className="thumbnail"
                style={
                  Object {
                    "backgroundImage": "url(undefined)",
                  }
                }
              />
              <span
                className="delete-image"
                onClick={[Function]}
              >
                ×
              </span>
            </div>
          </styled__ThumbnailPreview>
        </div>
      </styled__ThumbnailContainer>
    </div>
  </form>
</ChatRoomInput>
`;
